### PR TITLE
Task3 7

### PR DESCRIPTION
**Pull request by KatGaertner**
Added routing and profile view.

### CODE REVIEW

File: src/components/fav-button/fav-button.jsx [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+  const userData = JSON.parse(localStorage.getItem("userData"));
+  const token = localStorage.getItem("token");
+  const [isFavorited, setFavorited] = useState(false);
+
+  useEffect(() => {
+    if (userData.favorites.includes(movieID)) {
+      setFavorited(true);
+    } else {
+      setFavorited(false);
+    }
```
Comment (Stanley-Okwii) [inline src/components/fav-button/fav-button.jsx]: ⛏️You could refactor this  code like:
```suggestion
    const isFavoriteMovie = userData.favorites.includes(movieID);
    setFavorited(isFavoriteMovie);
```

File: src/components/login-view/login-view.jsx [hunk 2]
```diff
@@ -46,16 +49,17 @@ export const LoginView = ({ onLoggedIn }) => {
           placeholder="Username"
           value={username}
           onChange={(e) => setUsername(e.target.value)}
+          autoComplete="username"
```
Comment (Stanley-Okwii) [inline src/components/login-view/login-view.jsx:52]: 🔥🔥

File: src/components/movie-view/similar-movies.jsx [hunk 3]
```diff
@@ -0,0 +1,26 @@
+import PropTypes from "prop-types";
+import { moviesType } from "../../utils/types";
```
Comment (Stanley-Okwii) [inline src/components/movie-view/similar-movies.jsx:2]: Great use of utils 🎉

File: src/components/password-field/password-field.jsx [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+  autocomplete,
+}) => {
+  const [passwordFieldType, setPasswordFieldType] = useState("password");
+
+  const togglePasswordFieldType = () => {
+    if (passwordFieldType === "password") {
+      setPasswordFieldType("text");
+    } else {
+      setPasswordFieldType("password");
+    }
```
Comment (Stanley-Okwii) [inline src/components/password-field/password-field.jsx]: You could refactor this code like:
```suggestion
     const fieldType = passwordFieldType === "password" ? "password": "text";
      setPasswordFieldType(fieldType);
```

File: src/components/profile-view/layout.jsx [hunk 5]
```diff
@@ -0,0 +1,2 @@
+export const leftColumnWidth = 4;
```
Comment (Stanley-Okwii) [inline src/components/profile-view/layout.jsx]: ⛏️`.jsx` files are reserved for components, if your file does not have any components but merely constants, you can have this as `.js` file.

File: src/components/profile-view/profile-edit.jsx [hunk 6]
```diff
@@ -0,0 +1,146 @@
...
+  const [newUserData, setNewUserData] = useState({
+    name: userData.name,
+    email: userData.email,
+    birthday: userData.birthday ? userData.birthday.substring(0, 10) : "",
+    password: "",
+  });
+
+  const handleChange = (key, value) => {
+    const change = { [key]: value };
+    setNewUserData((newUserData) => ({ ...newUserData, ...change }));
```
Comment (Stanley-Okwii) [inline src/components/profile-view/profile-edit.jsx:22]: 🎉🔥

File: src/components/profile-view/profile-edit.jsx [hunk 7]
```diff
@@ -0,0 +1,146 @@
...
+  const handleSubmit = (event) => {
+    event.preventDefault();
+
+    let update = {};
+
+    if (event.target.reportValidity()) {
+      ["name", "email", "birthday"].forEach((key) => {
+        if (newUserData[key] && newUserData[key] !== userData[key]) {
+          update[key] = newUserData[key];
+          console.log("Changed " + key);
```
Comment (Stanley-Okwii) [inline src/components/profile-view/profile-edit.jsx]: Let's get rid of the trailing `console.logs`.

File: src/utils/links.jsx [hunk 8]
```diff
@@ -0,0 +1 @@
+export const API = 'https://movie-api-93299-83ca7447ffdb.herokuapp.com';
```
Comment (Stanley-Okwii) [inline src/utils/links.jsx]: 🥳

**Review by Stanley-Okwii — APPROVED** [review]
Amazing job, I appreciate the granular commits...

File: src/components/fav-button/fav-button.jsx (hunk 1 above)
Comment (KatGaertner) [reply src/components/fav-button/fav-button.jsx]: Thank you so much for your extensive feedback! I have implemented your suggestions.

**Pull request merged**